internal/core/docker_host_test.go

```diff
@@ -409,7 +446,28 @@ func TestExtractDockerSocketFromClient(t *testing.T) {
 
 		socket := extractDockerSocketFromClient(ctx, mockCli{OS: "Ubuntu"})
 
-		assert.Equal(t, "/this/is/a/sample.sock", socket)
+		require.Equal(t, "/this/is/a/sample.sock", socket)
+	})
+
+	t.Run("Unix Docker Socket is passed as docker.host property but not reachable", func(t *testing.T) {
+		content := "docker.host=" + DockerSocketSchema + "/this/is/a/sample.sock"
+		setupTestcontainersProperties(t, content)
+
+		t.Cleanup(resetSocketOverrideFn)
+
+		ctx := context.Background()
+		os.Unsetenv("TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE")
+		os.Unsetenv("DOCKER_HOST")
+
+		mockCallbackCheck(t, testCallbackCheckError)
+
+		var socket string
+		require.Panics(t, func() {
+			socket = extractDockerSocketFromClient(ctx, mockCli{OS: "Ubuntu"})
+		})
+
+		// the default Docker host without schema is returned because we cannot any other host
+		require.Equal(t, "", socket)
```

Comment:
suggestion: socket check is redundant here as the panics would have already failed.
```suggestion
		require.Panics(t, func() {
			extractDockerSocketFromClient(ctx, mockCli{OS: "Ubuntu"})
		})
```